### PR TITLE
[CHAT-26][CHAT-5] Send/verify sms codes Twillio

### PR DESCRIPTION
**Pull request by NaumVitanov**
* Added twillio. 
* Sending sms codes if not in dev env.
* New send sms endpoint.
* Created EnvironmentHelper to help identify the env. 
* Moved the appSettings in order to be accessible from the Application layer.

### CODE REVIEW

**Review by NikolaySlavov-96 — APPROVED** [review]
LGTM

File: ChatAPI.Application/UseCases/Implementations/UserService.cs [hunk 1]
```diff
@@ -1,41 +1,80 @@
...
+			if (EnvironmentHelper.IsDevelopment())
+				return Result.Success(HttpStatusCode.OK);
+
+			try
+			{
+				await VerificationResource.CreateAsync(appSettings.TwillioVerificationSid, dto.Phone, "sms");
+			}
+			catch (ApiException ex)
+			{
+				if (ex.Code == 60203)
```
Comment (vgechev) [inline ChatAPI.Application/UseCases/Implementations/UserService.cs]: Isn't it better for these to be constants?

File: ChatAPI.WebAPI/Extensions/ServiceCollectionExtension.cs [hunk 2]
```diff
@@ -146,5 +147,11 @@ public static IServiceCollection AddAuthenticationConfigured(this IServiceCollec
 							.WithSecurity(MessagePackSecurity.UntrustedData);
 					})
 				.Services;
+
+		public static IServiceCollection InitializeTwilio(this IServiceCollection service, IConfiguration configuration)
+		{
+			TwilioClient.Init(configuration["TwillioAccountSid"], configuration["TwillioAuthToken"]);
```
Comment (vgechev) [inline ChatAPI.WebAPI/Extensions/ServiceCollectionExtension.cs:153]: I think we can inject the class and have typization.

File: ChatAPI.Application/UseCases/Implementations/UserService.cs [hunk 3]
```diff
@@ -1,41 +1,80 @@
...
+					return Result.Failure("You can not send more codes at the moment. Please try again later.");
+
+				if (ex.Code == 60200)
+					return Result.Failure("Wrong phone number.");
+
+				return Result.Failure("Failed to send sms code.");
+			}
+
+			return Result.Success(HttpStatusCode.OK);
```
Comment (vgechev) [inline ChatAPI.Application/UseCases/Implementations/UserService.cs]: I don't think you need to specify the status code here.

**Review by vgechev — APPROVED** [review]
Good

**Pull request merged**